docs/sources/setup-grafana/configure-security/configure-authentication/okta/index.md

```diff
@@ -1,166 +1,146 @@
 
-1. Log in to the [Okta portal](https://login.okta.com/).
+## Integrate Okta OAuth2 with Grafana
 
-1. Go to Admin and then select **Developer Console**.
+To integrate your Okta OAuth2 provicer with Grafana using our Okta OAuth2 integration, follow these steps:
 
-1. Select **Applications**, then **Add Application**.
+1. [Create SWA app integrations](https://help.okta.com/en-us/Content/Topics/Apps/Apps_App_Integration_Wizard_SWA.htm) section.
 
-1. Pick **Web** as the platform.
+1. Set the callback URL for your OAuth2 app to `http://<my_grafana_server_name_or_ip>:<grafana_server_port>/login/generic_oauth`.
 
-1. Enter a name for your application (or leave the default value).
+   Ensure that the callback URL is the complete HTTP address that you use to access Grafana via your browser, but with the appended path of `/login/generic_oauth`.
 
-1. Add the **Base URI** of your application, such as https://grafana.example.com.
+   For the callback URL to be correct, it might be necessary to set the root_url option to [server], for example, if you are serving Grafana behind a proxy.
 
-1. Enter values for the **Login redirect URI**. Use **Base URI** and append it with `/login/okta`, for example: https://grafana.example.com/login/okta.
+1. Refer to the following table to update field values located in the `[auth.generic_oauth]` section of the Grafana configuration file:
 
-1. Click **Done** to finish creating the Okta application.
+   | Field                        | Description                                                                                                   |
+   | ---------------------------- | ------------------------------------------------------------------------------------------------------------- |
+   | `client_id`, `client_secret` | These values must match the client ID and client secret from your Okta OAuth2 app.                            |
+   | `auth_url`                   | The authorization endpoint of your OAuth2 provider. `https://<okta-tenant-id>.okta.com/oauth2v1authorize`     |
+   | `token_url`                  | The token endpoint of your Okta OAuth2 provider. `https://<okta-tenant-id>.okta.com/token`                    |
+   | `api_url`                    | The user information endpoint of your Okta OAuth2 provider. `https://<tenant-id>.okta.com/oauth2/v1/userinfo` |
+   | `enabled`                    | Enables Okta OAuth2 authentication. Set this value to `true`.                                                 |
 
-## Enable Okta OAuth in Grafana
+   Review the list of other Okta OAuth2 [configuration options]({{< relref "#configuration-options" >}}) and complete them, as necessary.
 
-1. Add the following to the [Grafana configuration file]({{< relref "../../../configure-grafana#configuration-file-location" >}}):
+1. Optional: [Configure a refresh token]({{< relref "#configure-a-refresh-token" >}}):
```

Comment:
Let's make these steps Okta specific.

Reply:
This references how to configure a refresh token section of the Okta documentation. Very little has changed from the original source and after testing it, it looks quite accurate. Please let me know if there's anything I'm missing.

Reply:
What I meant is:
* for step b. - can we specify the scope required by Okta instead of saying `the refresh token scope used by your OAuth2 provider`?
* for step c. - do you know if it is necessary to explicitly enable refresh tokens on Okta? Or are they enabled by default (like for GitLab)?

Reply:
Ah, I can see that you've already added this info in the `Configure a refresh token` [section](https://github.com/grafana/grafana/pull/71349/files#diff-8ace0d3ec49b661ceea540730d4c354e72ebc9492edca7a6b50cb536fad1e2eeR90).So I'd just mention that scope here as well.

Reply:
@IevaVasiljeva 

> for step b. - can we specify the scope required by Okta instead of saying the refresh token scope used by your OAuth2 provider?
I've added more descriptive steps in the following section `Configure a refresh token`

> for step c. - do you know if it is necessary to explicitly enable refresh tokens on Okta? Or are they enabled by default (like for GitLab)?
I can confirm they are not enabled by default at the Okta options.

Reply:
> Ah, I can see that you've already added this info in the Configure a refresh token section.So I'd just mention that scope here as well.

Added on both as requested